### PR TITLE
kubernetes: Adding openshift objects

### PR DESCRIPTION
**Pull request by petervo**

### CODE REVIEW

**Comment by stefwalter** [comment]
I think we should show the strategy:
 * Strategy: Rolling

Also for each trigger:
 * Trigger: ImageChange
 * Image: aoeu/nodejs-ex2:latest

And lastly replication settings:
 * Replicas: 5
 * Pod Selector: labels

File: pkg/kubernetes/client.js [hunk 1]
```diff
@@ -805,7 +808,7 @@ define([
         /* The watch objects we have open */
         self.watches = { "events": new KubernetesWatch("events", handle_event, handle_removed) };
         [ "nodes", "pods", "services", "replicationcontrollers",
-          "namespaces", "endpoints" ].forEach(self.include);
+          "namespaces", "endpoints", "deploymentconfigs" ].forEach(self.include);
```
Comment (stefwalter) [inline pkg/kubernetes/client.js]: Could we have the TopologyCtrl and DetailsCtrl do ```client.include("deploymentconfigs")```

They aren't used on every page after all...

File: pkg/kubernetes/details.css [hunk 2]
```diff
@@ -8,7 +8,8 @@
     top: -5px;
 }
 
-.listing-icons i.fa-server {
+.listing-icons i.fa-server,
+.listing-icons i.fa-gear {
```
Comment (stefwalter) [inline pkg/kubernetes/details.css]: Can we make the gear a little smaller? It's very dark soo 22px looks good.

File: pkg/kubernetes/topology-graph.css [hunk 3]
```diff
@@ -39,6 +39,10 @@ kubernetes-topology-graph {
     fill: #9467bd;
 }
 
+.kube-topology g.DeploymentConfig text {
+    font-family: FontAwesome;
```
Comment (stefwalter) [inline pkg/kubernetes/topology-graph.css]: Could you make this light gray? That's the color I used in the openshift web console for this.

File: pkg/kubernetes/views/topology-page.html [hunk 4]
```diff
@@ -61,6 +70,10 @@ <h3 translatable="yes">Node</h3>
       <circle r="15"></circle>
       <text y="7.5">&#xf1b8;</text>
     </g>
+    <g class="DeploymentConfig" id="vertex-DeploymentConfig">
+      <circle r="15"></circle>
+      <text y="7.5">&#xf013;</text>
```
Comment (stefwalter) [inline pkg/kubernetes/views/topology-page.html]: y = 8    font-size: 22px    fill: #888

**Comment by petervo** [comment]
Made changes except i didn't add pod selector. Pod selector only really makes sense at the deployment level not the deployment config level. We could pull the selector for the latest one. Or do we want to list deployments in a separate tab? It could include deployment status, reason and selectors. Maybe replace the template tab, i'm not sure that's of much use here. WDYT?

File: pkg/kubernetes/topology.js [hunk 5]
```diff
@@ -72,16 +76,26 @@ define([
...
+                    } else if (item.kind === "DeploymentConfig") {
+                        rels = client.select("ReplicationController", item.metadata.namespace,
+                                             {"openshift.io/deployment-config.name" : item.metadata.name});
+                    /* For Routes just build it out */
+                    } else if (item.kind === "Route" && item.spec.to) {
+                        var rel = client.lookup(item.spec.to.kind,
+                                                item.spec.to.name,
+                                                item.metadata.namespace);
+                        if (rel)
+                            rels[item.spec.to.kind + ":" + rel.metadata.uid] = rel;
```
Comment (stefwalter) [inline pkg/kubernetes/topology.js]: We now set a .key attribute on the items retrieved from the API server. Does that help here?

File: pkg/kubernetes/views/topology-page.html [hunk 6]
```diff
@@ -61,6 +70,14 @@ <h3 translatable="yes">Node</h3>
       <circle r="15"></circle>
       <text y="7.5">&#xf1b8;</text>
     </g>
+    <g class="DeploymentConfig" id="vertex-DeploymentConfig">
+      <circle r="15"></circle>
+      <text y="8">&#xf013;</text>
+    </g>
+    <g class="Route" id="vertex-Route">
```
Comment (stefwalter) [inline pkg/kubernetes/views/topology-page.html]: There is a pficon-route here: https://www.patternfly.org/styles/icons/ Is it available yet in our patternfly?

**Comment by stefwalter** [comment]
Could you add a simple integration test to ```./check-openshift``` to make sure that these show up in both the details, and topology views?

File: pkg/kubernetes/views/topology-page.html (hunk 6 above)
Comment (petervo) [inline pkg/kubernetes/views/topology-page.html]: No it's not

**Comment by petervo** [comment]
Added tests and fixups for newer changes that made it into master.

File: pkg/kubernetes/views/topology-page.html (hunk 6 above)
Comment (stefwalter) [inline pkg/kubernetes/views/topology-page.html]: I guess lets use fa-crosshairs cross hairs for now? Something similar in look.

**Comment by petervo** [comment]
Switched routes to fa-crosshairs

**Comment by petervo** [comment]
Updated to use pf-route now that we are on patternfly 2

**Comment by stefwalter** [comment]
@petervo Could you clear needswork when stuff is ready for review again?

**Comment by stefwalter** [comment]
Squashing. Test failure unrelated.

**Pull request closed without merging**